ZEP-12

artragis opened the pull request:
| Q | R |
| --- | --- |
| Correction de bugs ? | [oui] |
| Nouvelle Fonctionnalité ? | [oui] |
| Tickets (_issues_) concernés | [#2540, #2405, #2400, #2308, #2069, #1999, #1762, #1405, #2274, #2636, #1443, #1319, #1509, #1519, #1654, #1599, #1780, #850, #1226, #949, #2936, #2933, #2939, #2308, #2863, #2930] |

![](http://www.iab.net/media/image/countdown1.jpg)

Préparez les serveurs

![](http://thumbs.dreamstime.com/z/film-countdown-843869.jpg)

Rameutez les testeurs

![](https://jonjostcomingtoterms.files.wordpress.com/2012/08/film-leader-21.jpg)

Imaginez de nouveaux cours

![](http://www.cpr.org/sites/default/files/images/film-countdown-number-1-008.jpg)

@pierre-24 @DevHugo @vayel @GerardPaligot On vous aime

![](http://img0.gtsstatic.com/wallpapers/36d66a143906fdfcae55fd6881f1d156_large.jpeg)

Je ne peux pas donner d'instruction pour la QA. Tout le module de tuto et d'article est à tester.

Je reste à disposition.

Une citation du sage github : 

> This comparison is big!

DevHugo committed:
Premiere version de l'indexation du contenu

pierre-24 committed:
Ajoute T.U. : liste des validations

pierre-24 committed:
Ajoute T.U. : flux RSS

pierre-24 committed:
Merge pull request #192 from Vayel/gallery

Reformulation

pierre-24 committed:
Correction des `setUp()` pour rajouter les `UserGalleryFactory()`

pierre-24 committed:
Ajoute T.U. : `last_read_post()` et `first_unread_post()`

artragis committed:
donne des infos sur l'erreur 404 lors des redirections d'anciens tuto.

Commit:
Merge branch 'zep_12_double_stack' of https://github.com/artragis/zds-site into zep_12_double_stack

artragis committed:
met à jour la factory des extraits old pour qu'elle ait un bon commit message.

Commit:
Merge branch 'zep_12_double_stack' of https://github.com/artragis/zds-site into zep_12_double_stack

artragis committed:
Merge pull request #193 from pierre-24/zep_12_last_TU

Ajoute des T.U.'s

artragis committed:
gère les dates

Commit:
Merge branch 'zep_12_double_stack' of https://github.com/artragis/zds-site into zep_12_double_stack

artragis committed:
met à jour les dates dans les factories.

artragis committed:
exporte les catégories pour les articles.

Commit:
Merge branch 'zep_12_double_stack' of https://github.com/artragis/zds-site into zep_12_double_stack

pierre-24 committed:
Renvois un message d'erreur si slugs indentiques

artragis committed:
Merge pull request #197 from pierre-24/zep_12_slug_if_import

Renvois un message d'erreur si slugs indentiques

Commit:
Merge branch 'zep_12_double_stack' of https://github.com/artragis/zds-site into zep_12_double_stack

pierre-24 committed:
Ajoute une option pour desactiver le build des PDFs

DevHugo committed:
Application des remarques de artragis

pierre-24 committed:
Merge branch 'zep_12_double_stack' of https://github.com/artragis/zds-site into zep_12_slug_if_import

pierre-24 committed:
Edition de la documentation

pierre-24 committed:
Desactive le build des PDFs dans les tests (sauf quand utile)

pierre-24 committed:
Tout les auteurs suivent le topic de beta

pierre-24 committed:
Fait le lien entre le tuto et le sujet en beta

artragis committed:
Merge pull request #199 from pierre-24/zep_12_slug_if_import

Ajoute une option pour desactiver le build des PDFs

DevHugo committed:
Changement de la description

DevHugo committed:
Max_length dans url_to_redirect augmente

artragis committed:
Merge pull request #203 from pierre-24/zep_12_beta_authors

Améliore le comportement des bêta

Commit:
Merge branch 'zep_12_double_stack' of https://github.com/artragis/zds-site into zep_12_double_stack

artragis committed:
indicateurs munin

DevHugo committed:
Les chapitres s'affichaient comme des parties

DevHugo committed:
Suppression des tags

pierre-24 committed:
Suppression auteur et pas tuto si plusieurs auteurs

artragis committed:
Merge pull request #208 from pierre-24/zep_12_delete_when_multi_authors

Suppression de l'auteur et pas du tuto si il y a plusieurs auteurs

artragis committed:
flux rss intégrés

Commit:
Merge branch 'zep_12_double_stack' of https://github.com/artragis/zds-site into zep_12_double_stack

pierre-24 committed:
Repare possibilite de quitter la redaction d'un contenu

pierre-24 committed:
Edite T.U.

artragis committed:
Merge pull request #211 from pierre-24/zep_12_fix_authors_behavior

Répare possibilite de quitter la redaction d'un contenu

Commit:
Merge branch 'zep_12_double_stack' of https://github.com/artragis/zds-site into zep_12_double_stack

Commit:
merge les changements dans membres

DevHugo committed:
Daemonize thread

DevHugo committed:
Correction id commande

artragis committed:
corrige le bug de redirection lors de la suppression des auteurs

artragis committed:
fix le test

artragis committed:
règle le problème de clef.

artragis committed:
typo franglaise.

artragis committed:
règle le problème de changement de slug dans les tutos publiés.

artragis committed:
fixe le test

artragis committed:
refix le test.

DevHugo committed:
Affiche a la fin la commande Succefully indexed content

pierre-24 committed:
Corrige le test pour tenir compte du non-changement de slug

pierre-24 committed:
Tentative pour faire taire le test `test_last_reactions()`

pierre-24 committed:
Merge pull request #213 from pierre-24/zep_12_fix_test_slug_not_changed

Mini-fixs des tests

DevHugo committed:
Reindexation a la demande plutot qu'a la publication

DevHugo committed:
Correction du test

DevHugo committed:
Ajout des fichiers de migrations

DevHugo committed:
Suppression de l'ancien systeme d'indexation

DevHugo committed:
Correction de l'affichage de la date sur les sujets et posts

DevHugo committed:
Correction internationalisation

DevHugo committed:
Suppression des fixatures inutiles

DevHugo committed:
Ajout des migrations suite a l'internationalisation

yapper-git committed:
Amélioration de l'outil de diff

Situphen committed:
Permet à l'utilisateur de comparer deux versions de son tutoriel

artragis committed:
met à jour les instructions de mise à jour

Une histoire d'apocalypse de d'inception.

DevHugo committed:
Premiere version de la doc

yapper-git committed:
Ajout de tests unitaires pour DisplayHistory et DisplayDiff

artragis committed:
Merge pull request #214 from DevHugo/indexation_spacefox_copy_md

Indexation (avec la commande)

artragis committed:
Merge pull request #183 from Vayel/messages

Ajout de pulpe dans les messages

artragis committed:
bloque vraiment le renommage.

pierre-24 committed:
Indexe le manifest.json durant la migration

(et corrige erreurs UTF-8)

pierre-24 committed:
Corrige quelques erreurs

- Apparition des chemins de fichier
- Erreurs UTF-8
- "Pas de changements"

artragis committed:
Merge pull request #216 from pierre-24/zep_12_index_file

Indexe le manifest.json durant la migration

yapper-git committed:
Merge pull request #2 from pierre-24/minifix_diff

Corrige quelques erreurs

pierre-24 committed:
Merge pull request #215 from yapper-git/feat-htmldiff-zep12

Amélioration de l'outil de diff

artragis committed:
Que chacun s'éloigne de la piste de lancement

WinXaito commented:
Waouh quel PR .. Juste par curiosité, je dois prendre quel branch chez qui pour QA ?

Situphen commented:
D'après la PR c'est chez artragis, la branche `zep_12_double_stack`.

artragis commented:
notons qu'a url près tout est présent sur http://zds.francoisdambrine.me

Le 7 août 2015 15:10, Situphen notifications@github.com a écrit :

> D'après la PR c'est chez artragis, la branche zep_12_double_stack.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/zestedesavoir/zds-site/pull/2956#issuecomment-128695528
> .

DevHugo commented:
Comment on pourrais tester le script de migration avant de passer en pre-prod ?

artragis commented:
pour tester sur les vraies données, va falloir utiliser la preprod 
malheureusement.

Le 10/08/2015 20:06, Hugo Courtecuisse a écrit :

> Comment on pourrais tester le script de migration avant de passer en 
> pre-prod ?
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/zestedesavoir/zds-site/pull/2956#issuecomment-129551865.

DevHugo committed:
Procedure de mise à jours pour l'indexation

Ajoute une commande pour copier le contenu du dépot markdown publique vers le dossier
public-content/<nom-contenu>/extra_contents/<nom-contenu>. Cette copie est obligatoire pour avoir accés au markdown publié
aprés la publication.
Ajoute des précision sur la modification de la tache cron.

artragis committed:
Merge pull request #219 from DevHugo/upgrade_indexation

Procedure de mise à jours pour l'indexation

DevHugo commented:
Dommage, si par exemple, quelqu'un avec les accès de pré-prod, par-exemple, Eskimon ou firm1, (y'en a d'autre mais je connais pas tout ceux qui ont accès) fessait un dump de la base de pré-prod + copie des dossiers tutorials-public et tutorials-private, on pourrait mieux tester la migration ?

Peut-être que la QA serait plus facile avec un tableau récapitulatif de ce qu'il y'à a tester ? Découper en mini taches le problème permet toujours de se motiver (https://annuel.framapad.org/p/Zep12QA).

GerardPaligot commented:
Ne serait-il pas possible de tirer cette branche sur la preprod pour pouvoir tester la PR si nous avons besoin de données réelles ?

DevHugo commented:
On peut, mais il faudrait créé deux bases, une pour la pré-prod actuelle et une pour la zep-12 pour pouvoir être sur de ne pas impacter les données de la pré-prod. Ça inclue de modifier un peu le settings.py. 

C'est juste que la charge de travail est redirigé vers celui qui a accès à la pré-prod et pas à ceux qui ont dev la zep-12, si ça foire, celui qui à accès à la pré-prod devra refaire la manip à chaque test.

artragis commented:
> On peut, mais il faudrait créé deux bases, une pour la pré-prod actuelle et une pour la zep-12 pour pouvoir être sur de ne pas impacter les données de la pré-prod. Ça inclue de modifier un peu le settings.py. 

Pourquoi donc?

DevHugo commented:
Si ça foire, on sera bien content d'avoir une sauvegarde =) . Aprés y'a 80 % de coverage sur ce module donc, c'est limité comme risque.

Eskimon commented:
BOUM !

@artragis je te ferais un MP en copie avec @SpaceFox pour voir comment lancer un test en préprod directement.

artragis commented:
quand zds sera dispo :p?

2015-08-12 12:50 GMT+02:00 Eskimon notifications@github.com:

> BOUM !
> 
> @artragis https://github.com/artragis je te ferais un MP en copie avec
> @SpaceFox https://github.com/SpaceFox pour voir comment lancer un test
> en préprod directement.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/zestedesavoir/zds-site/pull/2956#issuecomment-130258915
> .

artragis commented:
Pourquoi "BUG"?

Situphen commented:
Car ça corrige des bugs, non ? (Des erreurs 500 et des trucs genre #1509)

DevHugo commented:
J'ai un peu tout testé (sauf la migration) et j'ai ouvert quelques tickets dans le dépôt d'Artragis.

artragis committed:
coquille dans les liste de catégories

artragis committed:
évite le bug en cas d'édition des commentaires

artragis commented:
Salut !
Il y a donc : 
- un problème d'intégration du nouveau style de la zep 03 (apporté par la zep 04)
- un problème lors de l'édition des commentaires
- l'archive zip des tutoriels publiés est incorrecte.

artragis committed:
BUg de l'édition des messages OK

artragis committed:
devrait corriger l'archive

+ quelques commentaires

artragis committed:
Le type de contenu est présélectionné lors de la création.

artragis committed:
fix le test des pubdates.

artragis committed:
oubli de refactorer les tests

DevHugo commented:
Je refais une QA ou j'attend que Situphen ferme les deux derniers tickets ?

Edit: Situphen à fait une PR, merci à lui.

Situphen committed:
Répare la page d'aide des contenus

artragis committed:
Merge pull request #229 from Situphen/zep-12

Répare la page d'aide des contenus

Commit:
corrige le message d'ajout en tant qu'auteur

Commit:
gère le cas de déplacement d'une partie unique

pierre-24 committed:
Corrige erreurs PEP-8

DevHugo commented:
Non testés: migration + indexation sinon le reste c'est ok, plus aucune idée de comment faire bugger le machin.

**Création de big-tutoriels**

| À QA | Résultat |
| --- | --- |
| Création d'un big tutoriel | O.K |
| Inporter un tutoriel | O.K |
| Création d'un big tutoriel avec le logo | O.K |
| Éditer le big-tutoriel (titre, description, introduction, conclusion, demande d'aide) | O.K |
| Éditer avec un message de commit | O.K |
| Ajouter un auteur | O.K |
| Gestion des auteurs (supprimer un auteur) | O.K |
| Mettre le sujet en beta, vérifier dans le forum, si le sujet existe et que les membres peuvent y acceder | O.K |
| Supprimer le tutoriel | O.K |
| Ajouter une partie | O.K |
| Éditer une partie | O.K |
| Déplacer une partie | O.K |
| Supprimer une partie | O.K |
| Ajouter un chapitre | O.K |
| Éditer un chapitre | O.K |
| Déplacer un chapitre | O.K |
| Supprimer un chapitre | O.K |
| Ajouter un extrait | O.K |
| Éditer un extrait | O.K |
| Déplacer un extrait | O.K |
| Supprimer un extrait | O.K |
| Demander la validation | O.K |
| Supprimer un extrait | O.K |
| Réservé et refuser le tutoriel | O.K |
| Valider le tutoriel | O.K |
| Dépublier le tutoriel | O.K |
| Filtres | O.K |
| Dépublier le tutoriel | O.K |

**Création de tutoriels**

| À QA | Résultat |
| --- | --- |
| Création d'un tutoriel | O.K |
| Inporter un tutoriel | O.K |
| Création d'un tutoriel avec le logo | O.K |
| Éditer le tutoriel (titre, description, introduction, conclusion, demande d'aide) | O.K |
| Éditer avec un message de commit | O.K |
| Ajouter un auteur | O.K |
| Gestion des auteurs (supprimer un auteur) | O.K |
| Mettre le sujet en beta, vérifier dans le forum, si le sujet existe et si n'importe qui peut acceder | O.K |
| Supprimer le tutoriel | O.K |
| Ajouter un chapitre | O.K |
| Éditer un chapitre | O.K |
| Déplacer un chapitre | O.K |
| Supprimer un chapitre | O.K |
| Ajouter un extrait | O.K |
| Éditer un extrait | O.K |
| Déplacer un extrait | O.K |
| Supprimer un extrait | O.K |
| Demmander la validation | O.K |
| Supprimer un extrait | O.K |
| Réservé et refuser le tutoriel | O.K |
| Valider le tutoriel | O.K |
| Dépublier le tutoriel | O.K |

**Création d'article**

| À QA | Résultat |
| --- | --- |
| Création d'un tutoriel | O.K |
| Inporter un tutoriel | O.K |
| Création d'un tutoriel avec le logo | O.K |
| Éditer le tutoriel (titre, description, introduction, conclusion, demande d'aide) | O.K |
| Éditer avec un message de commit | O.K |
| Ajouter un auteur | O.K |
| Gestion des auteurs (supprimer un auteur) | O.K |
| Mettre le sujet en beta, vérifier dans le forum, si le sujet existe et si n'importe qui peut acceder | O.K |
| Supprimer le tutoriel | O.K |
| Ajouter un chapitre | O.K |
| Éditer un chapitre | O.K |
| Déplacer un chapitre | O.K |
| Supprimer un chapitre | O.K |
| Ajouter un extrait | O.K |
| Éditer un extrait | O.K |
| Déplacer un extrait | O.K |
| Supprimer un extrait | O.K |
| Demmander la validation | O.K |
| Supprimer un extrait | O.K |
| Réservé et refuser le tutoriel | O.K |
| Valider le tutoriel | O.K |
| Dépublier le tutoriel | O.K |

**Création de commentaires pour les contenus**

| À QA | Résultat |
| --- | --- |
| Ajouter un commentaire sur chaque type de contenus | O.K |
| Effectuer les différentes opérations sur les messages (éditer, signaler) | O.K |
| Vérifier la pagination | O.K |

**Liste des tutoriels**

| À QA | Résultat |
| --- | --- |
| Vérifier que la liste des articles s'affiche bien | O.K |
| Vérifier que les catégories marchent bien | O.K |

**Liste des articles**

| À QA | Résultat |
| --- | --- |
| Vérifier que la liste des articles s'affiche bien | O.K |
| Vérifier que les catégories marchent bien | O.K |

**Historiques des versions**

| À QA | Résultat |
| --- | --- |
| Vérifier l'affichage | O.K |

*_Vérifier la ZEP 3 *_

| À QA | Résultat |
| --- | --- |
| Vérifier l'affichage | O.K |
| Tenter de changer les aides | O.K |

**Désincription**

| À QA | Résultat |
| --- | --- |
| Tuto publiés multi-auteurs | O.K |
| Tuto publiés unique auteur | O.K |

**Signaler une faute dans ce tutoriel**

| À QA | Résultat |
| --- | --- |
| Signaler la faute | O.K |
| Pas de bouton quand on est l'auteur | O.K |

**JS Fiddle**

| À QA | Résultat |
| --- | --- |
| Tester un article avec jsfiddle activé | O.K |
| Tester un tutoriel avec jsfiddle activé | O.K |
| Publié | O.K |

pierre-24 committed:
Merge pull request #234 from pierre-24/zep_12_fix_pep_xx

Corrige erreurs PEP-8

GerardPaligot commented:
Je voulais juste féliciter @DevHugo pour la QA impressionnante ! Bravo !

pierre-24 commented:
On doit aussi remercier  @Vayel, qui a fait énormément de QA en continu et qui est responsable d'un grand nombre de ces "OK" :)

Situphen committed:
Personnalise plus les pages listant les articles et tutoriels

DevHugo committed:
Corrige le souci qui empéchait la mise à jours des index de recherche

artragis committed:
Merge pull request #236 from Situphen/zep-12

Personnalise plus les pages listant les articles et tutoriels

artragis committed:
Merge pull request #235 from DevHugo/correction_indexation

Corrige le souci qui empéchait la mise à jours des index de recherche

DevHugo committed:
Exclus les contenus de redirection de l'indexation

artragis committed:
Merge pull request #237 from DevHugo/indexation_exclude_redirect

Exclus les contenus de redirection de l'indexation

pierre-24 committed:
Corrige l'incohrence des slugs

artragis committed:
Merge pull request #243 from pierre-24/zep_12_fix_slugs

Corrige l'incohérence des slugs

pierre-24 committed:
Corrige date des anciens commentaires

pierre-24 committed:
Merge pull request #245 from pierre-24/zep_12_fix_pubdate

Corrige la date des anciens commentaires

DevHugo committed:
Augmentation de la limite pour les titres des extraits, des conteners

artragis committed:
Merge pull request #249 from DevHugo/update_max_length_title

Augmentation de la limite pour les titres des extraits, des conteners

artragis committed:
correction du update.md suite à la première expérience de migration.

artragis committed:
dépublication et migration, tout sha doit être à sa place

artragis committed:
Devrait régler le problème des notifications

artragis committed:
évite de migrer des notes qui n'existent pas

artragis committed:
mise à jour de l'historique de validation

DevHugo committed:
During zep12 migration, copy image from none directory to galery

artragis committed:
Merge pull request #250 from DevHugo/convert_image_in_none_directory

During zep12 migration, copy image from none directory to galery

DevHugo committed:
Correction pep8

pierre-24 committed:
Merge pull request #251 from DevHugo/correction_pep8

Correction pep8

DevHugo commented:
Rien dit

DevHugo committed:
Copy thumbails from article to galleries

artragis committed:
Merge pull request #252 from DevHugo/copy_file_from_articles_to_galleries

Copy thumbails from article to galleries

artragis committed:
permet d'éviter les probl-mes de validation sur les articles

+ un petit bug sans ticket dans le unpublish

artragis committed:
légère typo dans le label de la validation.

artragis committed:
devrait corriger le bug des déplacement.

artragis committed:
devrait corriger le bug des déplacement après un autre enfant.

artragis committed:
fix le test en mettant à jour l'objet.

artragis committed:
l'article et son objet public n'étaient pas liés dans le test.

DevHugo committed:
Ajout de quelques explications

artragis committed:
corrige le bug d'affichage de la pagination

artragis committed:
enlève les notifications parasite.

DevHugo committed:
Continue indexing content after an error

pierre-24 committed:
Merge pull request #261 from DevHugo/continue_indexing_after_error

Continue d'indexer aprés une erreur

pierre-24 committed:
Merge pull request #259 from DevHugo/ameliorations_mise_jours_info_update

Ajout de quelques explications dans le update.md

pierre-24 committed:
Inverse le slug sur le tutoriel mappé et exporté

DevHugo committed:
Corrige l'affichage des dates dans les resultats de la recherche

pierre-24 committed:
Merge pull request #264 from DevHugo/correction_indexation_date

Corrige l'affichage des dates dans les resultats de la recherche

artragis committed:
Merge pull request #263 from pierre-24/zep_12_le_bon_slug_au_bon_endroit

Inverse le slug sur le tutoriel mappé et exporté

DevHugo committed:
Supprime un warning, si la date de publication n'est pas specifie

DevHugo committed:
Don't display manifest in diff

artragis committed:
corrige les licences

artragis committed:
corrige le titre des mp.

DevHugo committed:
Default content per page is now 42

pierre-24 committed:
Merge pull request #280 from DevHugo/default_content_per_page

Le nombre de contenu par page passe à 42

pierre-24 committed:
Merge pull request #278 from DevHugo/dont_display_manifest

N'affiche plus le manifest dans les diffs

pierre-24 committed:
Merge pull request #277 from DevHugo/delete_warning_if_pubdate_is_not_specified

Supprime un warning, si la date de publication n'est pas specifie

pierre-24 committed:
Utilise `subprocess.call()` à la place de `os.system()` pour éviter
les `os.chdir()`

pierre-24 committed:
Modifie `update.md`

pierre-24 committed:
Fait de même pour `generate_pdf`

artragis committed:
Merge pull request #282 from pierre-24/zep_12_no_chdir

Utilise subprocess pour éviter le changement de dossier

artragis committed:
Merge pull request #283 from pierre-24/zep_12_right_to_staff

Modifie update.md

pierre-24 committed:
Indèxe via l'archive ZIP au lieu du dépot

pierre-24 committed:
Correction documentation et options

pierre-24 committed:
Correction des tests

pierre-24 committed:
Améliore la documentation du module de recherche

pierre-24 committed:
Améliore l'output de la commande

artragis committed:
Merge pull request #290 from pierre-24/zep_12_index_with_zip

Indèxe via l'archive ZIP au lieu du dépot

Situphen committed:
Quelques corrections dans la documentation et le fichier `update.md`

pierre-24 committed:
Corrige l'utilisation des permissions dans le code

artragis committed:
Merge pull request #295 from pierre-24/zep_12_fix_permissions

Corrige l'utilisation des permissions dans le code

artragis committed:
permet l'import des anciens tuto

pierre-24 committed:
Merge pull request #297 from artragis/dot-tuto-correction

permet l'import des anciens tuto

artragis committed:
Fix le comptage sur la HP.

firm1 commented on tox.ini:
```diff
@@ -19,6 +19,15 @@ commands =
        coverage run --source='.' {toxinidir}/manage.py test {posargs}
        flake8
 
+[testenv:back_mysql_tuto]
```
Y'a t-il une différence entre `back_mysql_tuto` et `back_mysql` ? Si oui laquelle ?

pierre-24 replied:
Oui. On a re-séparé les tests des tutos et les autres. Pas de problèmes juqu'ici (et ça fait 3 mois)

Situphen replied:
@pierre-24 : Oui mais pourquoi créer `back_mysql` et `back_mysql_tuto` ? Vous auriez pu utiliser deux fois `back_mysql` ?

pierre-24 replied:
Euuuuh ... @artragis  ?

artragis replied:
non c'est un test que j'ai pas poussé et qui pollue le truc. Par contre 
je fais plus de magie noire sur tox, pour y regarder faudra quelqu'un 
qui s'y connait.

Le 26/08/2015 20:09, Pierre Beaujean a écrit :

> In tox.ini 
> https://github.com/zestedesavoir/zds-site/pull/2956#discussion_r38014782:
> 
> > @@ -19,6 +19,15 @@ commands =
> >         coverage run --source='.' {toxinidir}/manage.py test {posargs}
> >         flake8
> > 
> > +[testenv:back_mysql_tuto]
> 
> Euuuuh ... @artragis https://github.com/artragis ?
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/zestedesavoir/zds-site/pull/2956/files#r38014782.

pierre-24 replied:
Corrigé.

firm1 replied:
Cette brique n'est pas toujours corrigée. Il faut supprimer tout le bloc `[testenv:back_mysql_tuto]` ainsi que sa définition car il est en doublon de `[testenv:back_mysql]`.

Situphen replied:
+1

firm1 commented on update.md:
```diff
@@ -280,3 +280,57 @@ La recherche est maintenant en français:
+
+et de les écarter temporairement (en les déplacant dans un autre dossier), afin de ne pas géner la migration. Ils devront être ré-importé manuellement dans le futur, si cela est possible.
+
+**Migrations des dépendences, de la base de données et des contenus**
```
Peut-être préciser que cette partie doit être faite après déploiement automatique de la zep12 et que les tutos seront indisponibles entre le déploiement auto et cette partie ? Peut-être même qu'il serait judicieux de passer le site en maintenance le temps de la migration du contenu ?

pierre-24 replied:
Ça me semblais évident, mais si tu veux qu'on le précise :)

pierre-24 replied:
Non précisé, trop évident.

firm1 commented on update.md:
```diff
@@ -280,3 +280,57 @@ La recherche est maintenant en français:
+
+**Migrations des dépendences, de la base de données et des contenus**
+
+- De nouvelles dépendances ont été ajoutées (*beautifulsoup4*, *uuslug*). Pour installer ces nouvelles dépendences, utilisez la commande  `pip install -r requirements.txt`.
```
ceci est est normalement réalisé par le script de déploiement

firm1 commented on update.md:
```diff
@@ -280,3 +280,57 @@ La recherche est maintenant en français:
+**Migrations des dépendences, de la base de données et des contenus**
+
+- De nouvelles dépendances ont été ajoutées (*beautifulsoup4*, *uuslug*). Pour installer ces nouvelles dépendences, utilisez la commande  `pip install -r requirements.txt`.
+- La base de données a été modifiée (ajout de nouvelles tables pour les contenus, ainsi que des tables pour la recherche). Pour installer ces nouvelles tables, utilisez la commande `python manage.py migrate`.
```
ceci est est normalement réalisé par le script de déploiement

firm1 commented on update.md:
````diff
@@ -280,3 +280,57 @@ La recherche est maintenant en français:
+  ```bash
+  DB="dbname";USER="dbusername";PASS="dbusername";mysql "$DB" --host=127.0.0.1 -e "SHOW TABLES" --batch --skip-column-names -u $USER -p=$PASS| xargs -I{} echo 'ALTER TABLE '{}' CONVERT TO CHARACTER SET utf8 COLLATE utf8_general_ci;'  | mysql "$DB" --host=127.0.0.1  -u $USER -p=$PASS
+  ```
+- La génération des PDF par Pandoc, peut-être très longue, il est fortement conseillé de désactiver temporairement la génération de PDF . Pour cela, dans le fichier settings_prod.py, passer la variable `ZDS_APP['content']['build_pdf_when_published']` à `False`.
````
Il faudrait préciser ici de relancer l'app pour que le paramètre soit pris en compte.

firm1 commented on update.md:
```diff
@@ -280,3 +280,57 @@ La recherche est maintenant en français:
+- La génération des PDF par Pandoc, peut-être très longue, il est fortement conseillé de désactiver temporairement la génération de PDF . Pour cela, dans le fichier settings_prod.py, passer la variable `ZDS_APP['content']['build_pdf_when_published']` à `False`.
+- Il faut maintenant migrer tous les contenus, pour cela utiliser la commande `python manage.py migrate_to_zep12`. Cette commande peut prendre plusieurs minutes.
+- Lancer la génération des PDF avec la commande: `python manage.py generate_pdf`, si vous désirez que les erreurs générées soient loggées, envoyez la sortie standard vers le fichier de votre choix. Prendre soin d'avoir activé le log de Pandoc.
+- Repasser la variable `ZDS_APP['content']['build_pdf_when_published']` à `True`.
```
Il faudrait préciser ici de relancer l'app pour que le paramètre soit pris en compte.

firm1 commented on update.md:
```diff
@@ -280,3 +280,57 @@ La recherche est maintenant en français:
+
+**Migrations du module de recherche**
+
+- Arrêter Solr : `supervisorctl stop solr`
```
A confirmer si on le nouveau serveur de prod utilise supervisor ou systemd

pierre-24 replied:
Je n'en ai aucune foutue idée, et jusqu'ici, personne ne sais à part @SpaceFox ;)

SpaceFox replied:
systemd.

artragis replied:
comment on stop un service via systemd du coup?

firm1 replied:
`systemctl stop mon-service` ?

firm1 commented on update.md:
```diff
@@ -280,3 +280,57 @@ La recherche est maintenant en français:
+- Regénérer le schema.xml avec la commande: `python manage.py build_solr_schema > /votre/path/vers/solr-4.9.1/example/solr/collection1/conf/schema.xml`
+- Recopier les dépots markdown dans le contenu public: `python manage.py index_content --copy-repository`
+- Changer la tache cron qui perment d'indexer les contenus de: `python manage.py update_index` à `python manage.py index_content --only-flagged >> /var/log/indexation.txt && python manage.py update_index >> /var/log/indexation.txt`
+- Créé une autre tache cron pour supprimer réguliérement le fichier de log /var/log/indexation.txt.
```
Plutôt mettre en place une rotation de ces logs ?

pierre-24 replied:
?

firm1 commented on update.md:
```diff
@@ -280,3 +280,57 @@ La recherche est maintenant en français:
+- Démarrer Solr : `supervisorctl start solr`
+- Indexer le contenu avec Solr: `python manage.py rebuild_index`. Attention, cette commande peut prendre plusieurs minutes.
+
+**Aprés la migration**
```
`Aprés` => `Après` :)

firm1 commented on update.md:
```diff
@@ -280,3 +280,57 @@ La recherche est maintenant en français:
+
+**Aprés la migration**
+
+- Par défaut la pagination est mise à 42 éléments, mais nous affichons 2 ou 3 colonnes selon les largeurs d'écran. Pour le changer, il faut modifier la variable `ZDS_APP['content']['content_per_page']` dans le settings_prod.py
```
Il faudrait préciser ici de relancer l'app pour que le paramètre soit pris en compte.

artragis committed:
évite les doublons de old_pk

artragis committed:
migre le canonical

pierre-24 committed:
Met le lien vers le message avec la citation

firm1 commented on zds/settings.py:
```diff
@@ -329,7 +332,7 @@
 PANDOC_LOC = ''
 PANDOC_PDF_PARAM = ("--latex-engine=xelatex "
-                    "--template=../../assets/tex/template.tex -s -S -N "
+                    "--template=../../../assets/tex/template.tex -s -S -N "
```
Travis à l'air de crasher beaucoup d'erreur lié à la non trouvaille de ce template.

pierre-24 replied:
Et pour cause, c'est le build des PDF des anciens tutos qui plante systématiquement parce que le dossier de build a descendu d'un niveau. Mais ne t'inquiète pas, on a un test pour vérifier que les PDFs sont bien construits, donc on est bon :)

firm1 replied:
Du coup pourquoi conserver le build des anciens tutos si on utilise celui des nouveaux tutos maintenant ? On ne devrait pas virer carrément l'ancien dossier [tutorial](https://github.com/artragis/zds-site/tree/zep_12_double_stack/zds/tutorial) en lieu et place de tutorialv2 ?

pierre-24 replied:
Parce que je _flippe_ total d'avoir fait une connerie et que donc, je préfère garder l'ancienne _stack_ en stock le temps drêtre sur que tout est OK (toujours accessible en partie via `/oldtutoriels/` et `/oldarticles/`, mais je le crie pas trop fort pour que les auteurs ne fasse pas n'importe quoi). De plus, le module est au moins nécessaire pour la migration.

artragis replied:
Non, on en a besoin pour la migration.
Par contre, un jour, ça sera supprimé, oui. Notamment quand il faudra
passer à python3.

2015-08-28 11:36 GMT+02:00 firm1 notifications@github.com:

> In zds/settings.py
> https://github.com/zestedesavoir/zds-site/pull/2956#discussion_r38184381
> :
> 
> > @@ -329,7 +332,7 @@
> > 
> >  PANDOC_LOC = ''
> >  PANDOC_PDF_PARAM = ("--latex-engine=xelatex "
> > -                    "--template=../../assets/tex/template.tex -s -S -N "
> > -                    "--template=../../../assets/tex/template.tex -s -S -N "
> 
> Du coup pourquoi conserver le build des anciens tutos si on utilise celui
> des nouveaux tutos maintenant ? On ne devrait pas virer carrément l'ancien
> dossier tutorial
> https://github.com/artragis/zds-site/tree/zep_12_double_stack/zds/tutorial
> en lieu et place de tutorialv2 ?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/zestedesavoir/zds-site/pull/2956/files#r38184381.

pierre-24 replied:
On fera une PR post-ZEP-12 qui éliminera totalement test, module et fichiers.

(NB: si on l'enlève des tests, coverage chute).

firm1 replied:
hmm, je ne suis pas certain que tous les fichiers (peut-être les models, mais ptet pas les views) soient necessaires pour la migration. Après si c'est pour garder une solution de repli au cas ou, pourquoi pas. C'est juste que du coup on est obligé de faire cohabiter deux modules en parallèle (ainsi que les paramètres qui vont avec).

Faites tout de même attention au SEO, car les urls `oldarticles` et `oldtutoriels` risquent de faire doublon avec les urls `articles` et `tutoriels`. On a un risque de duplicate content interne sur 2 urls.

pierre-24 replied:
Les vues sont nécéssaire dans le cas ou une migration se serait mal passé et qu'on devrait faire du copier/coller bourin à la main.

Par contre, on devrait blacklister les urls dans le robots.txt, là je suis d'accord.

artragis replied:
> Faites tout de même attention au SEO, car les urls oldarticles et
> oldtutoriels risquent de faire doublon avec les urls articles et tutoriels.
> On a un risque de duplicate content interne sur 2 urls.

encore faudrait-il qu'il y ait une trace de ces urls quelques part.

Le 28 août 2015 12:04, Pierre Beaujean notifications@github.com a écrit :

> In zds/settings.py
> https://github.com/zestedesavoir/zds-site/pull/2956#discussion_r38186319
> :
> 
> > @@ -329,7 +332,7 @@
> > 
> >  PANDOC_LOC = ''
> >  PANDOC_PDF_PARAM = ("--latex-engine=xelatex "
> > -                    "--template=../../assets/tex/template.tex -s -S -N "
> > -                    "--template=../../../assets/tex/template.tex -s -S -N "
> 
> Les vues sont nécéssaire dans le cas ou une migration se serait mal passé
> et qu'on devrait faire du copier/coller bourin à la main.
> 
> Par contre, on devrait blacklister les urls dans le robots.txt, là je suis
> d'accord.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/zestedesavoir/zds-site/pull/2956/files#r38186319.

firm1 replied:
> encore faudrait-il qu'il y ait une trace de ces urls quelques part. 

Ces urls (les olds) sont exposées partout dans le [sitemap.xml](https://beta.zestedesavoir.com/sitemap-articles.xml), donc Google les verra à tous les coups.

artragis replied:
Eh merde, c'est pas normal. faut les enlever du sitemap.

Le 28 août 2015 12:14, firm1 notifications@github.com a écrit :

> In zds/settings.py
> https://github.com/zestedesavoir/zds-site/pull/2956#discussion_r38187086
> :
> 
> > @@ -329,7 +332,7 @@
> > 
> >  PANDOC_LOC = ''
> >  PANDOC_PDF_PARAM = ("--latex-engine=xelatex "
> > -                    "--template=../../assets/tex/template.tex -s -S -N "
> > -                    "--template=../../../assets/tex/template.tex -s -S -N "
> 
> encore faudrait-il qu'il y ait une trace de ces urls quelques part.
> 
> Ces urls (les olds) sont exposées partout dans le sitemap.xml
> https://beta.zestedesavoir.com/sitemap-articles.xml, donc Google les
> verra à tous les coups.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/zestedesavoir/zds-site/pull/2956/files#r38187086.

firm1 replied:
En tout cas si vous voulez conserver les deux applications (v1 et v2) active, il faut (comme vous l'avez fait pour les autres paramètres du settings) dupliquer ce paramètre `PANDOC_PDF_PARAM` surtout si ça valeur differe entre la v1 et la v2 pour que ce soit correct.

artragis replied:
Non, il n'y a pas de raison qu'on publie un tuto avec l'ancienne version.
Pour l'instant on la garde à cause des tests et par sécurité, mais si
demain ça pète bah au moins on a un rollback. Notons qu'on a de toute façon
besoin des modèles.

Le 28 août 2015 13:55, firm1 notifications@github.com a écrit :

> In zds/settings.py
> https://github.com/zestedesavoir/zds-site/pull/2956#discussion_r38192440
> :
> 
> > @@ -329,7 +332,7 @@
> > 
> >  PANDOC_LOC = ''
> >  PANDOC_PDF_PARAM = ("--latex-engine=xelatex "
> > -                    "--template=../../assets/tex/template.tex -s -S -N "
> > -                    "--template=../../../assets/tex/template.tex -s -S -N "
> 
> En tout cas si vous voulez conserver les deux applications (v1 et v2)
> active, il faut (comme vous l'avez fait pour les autres paramètres du
> settings) dupliquer ce paramètre PANDOC_PDF_PARAM surtout si ça valeur
> differe entre la v1 et la v2 pour que ce soit correct.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/zestedesavoir/zds-site/pull/2956/files#r38192440.

firm1 commented on zds/settings.py:
```diff
@@ -448,7 +451,24 @@
         'home_number': 5,
-        'helps_per_page': 20
+        'helps_per_page': 20,
+        'content_per_page': 42,
```
Y'a t-il une différence entre ce `content_per_page` et celui qui se trouve sous la clé `content` plus bas ? Si oui, quelle est cette différence ? et pourquoi le même nom ?

pierre-24 replied:
Parce qu'on ne se sert plus de `ZDS_APP['tutorial']`, donc il y a des doublons le temps qu'on supprime les anciens modules (pas tout de suite ^^)

firm1 commented on zds/settings.py:
```diff
@@ -448,7 +451,24 @@
         'default_license_pk': 7,
         'home_number': 5,
-        'helps_per_page': 20
+        'helps_per_page': 20,
```
idem pour le `helps_per_page`, ne fait-il pas doublon avec celui dans `content` ?

firm1 commented on zds/tutorial/search_indexes.py:
```diff
@@ -1,107 +1,106 @@
 # coding: utf-8
 
-from django.db.models import Q
+# from django.db.models import Q
```
Cette remarque je la mets ici, mais elle est valable pour tout le fichier. ça ne sert à rien de committer du code commenté. Si jamais on veut le retrouver git est notre ami, mais committer du code commenté .... autant le virer tout simplement.

pierre-24 replied:
On a du ne pas faire attention une fois ^^

firm1 commented on zds/search/utils.py:
```diff
@@ -0,0 +1,273 @@
+    bs = BeautifulSoup(html)
+
+    keywords = u''
+    for tag in bs.findAll(['h1', 'h2', 'h3', 'i', 'strong']):
```
la balise `i` n'est pas utilisée par le parseur de zds. Tu devrais la remplacer par la balise `em` qui est la balise html correspondant à l'emphase `*mon-texte*`.

pierre-24 replied:
ping @DevHugo :)

firm1 commented on zds/member/tests/tests_views.py:
```diff
@@ -29,9 +27,8 @@
-overrided_zds_app['tutorial']['repo_public_path'] = os.path.join(BASE_DIR, 'tutoriels-public-test')
-overrided_zds_app['article']['repo_path'] = os.path.join(BASE_DIR, 'article-data-test')
+overrided_zds_app['content']['repo_private_path'] = os.path.join(BASE_DIR, 'contents-private-test')
+overrided_zds_app['content']['repo_public_path'] = os.path.join(BASE_DIR, 'contents-public-test')
```
Il faut penser à appliquer la modification dans la fonction [teardown](https://github.com/artragis/zds-site/blob/zep_12_double_stack/zds/member/tests/tests_views.py#L899) pour nettoyer les bon répertoire à la fin du test

pierre-24 replied:
Fait.

firm1 commented on zds/member/tests/tests_models.py:
```diff
@@ -11,21 +11,19 @@
-overrided_zds_app['tutorial']['repo_public_path'] = os.path.join(BASE_DIR, 'tutoriels-public-test')
-overrided_zds_app['article']['repo_path'] = os.path.join(BASE_DIR, 'article-data-test')
+overrided_zds_app['content']['repo_private_path'] = os.path.join(BASE_DIR, 'contents-private-test')
+overrided_zds_app['content']['repo_public_path'] = os.path.join(BASE_DIR, 'contents-public-test')
```
Penser à appliquer la modification dans le tearDown plus bas afin de bien nettoyer les bon répertoires

pierre-24 replied:
Fait.

firm1 commented on zds/gallery/views.py:
```diff
@@ -165,17 +188,27 @@ def modify_gallery(request):
 
-        perms = UserGallery.objects.filter(gallery__pk__in=free_galleries,
-                                           user=request.user, mode="W").count()
+        perms = UserGallery.objects.filter(gallery__pk__in=free_galleries, user=request.user, mode="W").count()
```
le `W` devrait être constanté quelque part.

pierre-24 replied:
Je suis d'accord. Ceci étant dit, c'est _hors-scope_.

artragis replied:
En fait la constante existe déjà.

2015-08-27 8:27 GMT+02:00 Pierre Beaujean notifications@github.com:

> In zds/gallery/views.py
> https://github.com/zestedesavoir/zds-site/pull/2956#discussion_r38065711
> :
> 
> > ```
> >          else:
> >              free_galleries.append(g_pk)
> > ```
> > -        perms = UserGallery.objects.filter(gallery__pk__in=free_galleries,
> > -                                           user=request.user, mode="W").count()
> > -        perms = UserGallery.objects.filter(gallery__pk__in=free_galleries, user=request.user, mode="W").count()
> 
> Je suis d'accord. Ceci étant dit, c'est _hors-scope_.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/zestedesavoir/zds-site/pull/2956/files#r38065711.

firm1 commented on zds/article/search_indexes.py:
```diff
@@ -1,28 +1,28 @@
-
-    def prepare_permissions(self, obj):
-        return "public"
+# from haystack import indexes
```
Un fichier de commentaire n'a pas à être commité quand on utilise un gestionnaire de version fait pour ça

firm1 commented on .travis.yml:
```diff
@@ -7,11 +7,10 @@ python:
 
 env:
-  global:
-    - secure: "azmDZZQZzf88zpbkYpLpxI66vpEVyv+kniW0QdWAt4qlys8P5OcO3VJBR5YT85vlvnjN9b6raWQAL1ymee0WmVdTmzXed8XjZv7t9QXVw7pfezxMKlEftVp/4Cu4wtvbew0ViZXNWV2hNXHzEqlhgnoIOq94i0UzZ7grMrI0xm0="
```
pourquoi avoir fait sauté ce `secure` ? il me semble qu'il est utile pour le bot de build

artragis replied:
magie noire travis et tox. Faut demander aux experts.

2015-08-27 10:10 GMT+02:00 firm1 notifications@github.com:

> In .travis.yml
> https://github.com/zestedesavoir/zds-site/pull/2956#discussion_r38071319
> :
> 
> > @@ -7,11 +7,10 @@ python:
> > - 2.7
> >   
> >   env:
> >   -  global:
> >   -    - secure: "azmDZZQZzf88zpbkYpLpxI66vpEVyv+kniW0QdWAt4qlys8P5OcO3VJBR5YT85vlvnjN9b6raWQAL1ymee0WmVdTmzXed8XjZv7t9QXVw7pfezxMKlEftVp/4Cu4wtvbew0ViZXNWV2hNXHzEqlhgnoIOq94i0UzZ7grMrI0xm0="
> 
> pourquoi avoir fait sauté ce secure ? il me semble qu'il est utile pour
> le bot de build
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/zestedesavoir/zds-site/pull/2956/files#r38071319.

firm1 replied:
il faut juste mettre les TEST_APP="blabla" à la place de ceux existant, sans casser la structure générale. Sinon, ça créera une regression

artragis replied:
je te laisse faire, tu as le droit de faire une PR sur mon dépôt tu sais.

2015-08-27 10:50 GMT+02:00 firm1 notifications@github.com:

> In .travis.yml
> https://github.com/zestedesavoir/zds-site/pull/2956#discussion_r38074295
> :
> 
> > @@ -7,11 +7,10 @@ python:
> > - 2.7
> >   
> >   env:
> >   -  global:
> >   -    - secure: "azmDZZQZzf88zpbkYpLpxI66vpEVyv+kniW0QdWAt4qlys8P5OcO3VJBR5YT85vlvnjN9b6raWQAL1ymee0WmVdTmzXed8XjZv7t9QXVw7pfezxMKlEftVp/4Cu4wtvbew0ViZXNWV2hNXHzEqlhgnoIOq94i0UzZ7grMrI0xm0="
> 
> il faut juste mettre les TEST_APP="blabla" à la place de ceux existant,
> sans casser la structure générale. Sinon, ça créera une regression
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/zestedesavoir/zds-site/pull/2956/files#r38074295.

firm1 replied:
github, fork, syncho, toussa ... :p

pierre-24 replied:
Si c'est une modif "rapide", tu peux éditer en ligne ;)

(surtout que ni moi ni artragis ne maitrisons vraiment Tox)

firm1 replied:
Justement je ne peux pas éditer en ligne, mon fork n'est pas synchro. Mais en gros, il faudrait remplacer : 

``` yaml
env:
  - TEST_APP="-e back_mysql -- zds.member zds.mp zds.utils zds.forum zds.gallery zds.pages zds.search"
  - TEST_APP="-e back_mysql_tuto -- zds.article zds.tutorial zds.tutorialv2"
  - TEST_APP="-e front"
```

par : 

``` yaml
env:
  global:
    - secure: "azmDZZQZzf88zpbkYpLpxI66vpEVyv+kniW0QdWAt4qlys8P5OcO3VJBR5YT85vlvnjN9b6raWQAL1ymee0WmVdTmzXed8XjZv7t9QXVw7pfezxMKlEftVp/4Cu4wtvbew0ViZXNWV2hNXHzEqlhgnoIOq94i0UzZ7grMrI0xm0="
  matrix:
    - TEST_APP="-e back_mysql -- zds.member zds.mp zds.utils zds.forum zds.gallery zds.pages zds.search"
    - TEST_APP="-e back_mysql_tuto -- zds.article zds.tutorial zds.tutorialv2"
    - TEST_APP="-e front"
```

firm1 commented on assets/scss/pages/_tutorial-history.scss:
```diff
@@ -0,0 +1,3 @@
+.commits-compare-form button {
+    float: none !important;
```
ah les `important` c'est pas joli :)

artragis replied:
demande @sandhose et @situphen

2015-08-27 10:17 GMT+02:00 firm1 notifications@github.com:

> In assets/scss/pages/_tutorial-history.scss
> https://github.com/zestedesavoir/zds-site/pull/2956#discussion_r38071835
> :
> 
> > @@ -0,0 +1,3 @@
> > +.commits-compare-form button {
> > -    float: none !important;
> 
> ah les important c'est pas joli :)
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/zestedesavoir/zds-site/pull/2956/files#r38071835.

Situphen replied:
Des fois, on n'a pas le choix...

firm1 commented on doc/source/front-end/template-tags.rst:
```diff
@@ -505,3 +480,73 @@ Par exemple, le code suivant affichera le lien vers le forum uniquement si celui
+=========================
+
+Ce module défini un *templatetag* utilisé dans le module de tutoriel (v2) dans le but de générer la hiérarchie des tutos et l'arbre
+des déplacements possibles d'un élément. Il s'agint d'un wrapper autour de ``zds.tutorialv2.utils.get_target_tagged_tree``.
```
`Il s'agint` => `Il s'agit`

firm1 commented on doc/source/front-end/template-tags.rst:
```diff
@@ -505,3 +480,73 @@ Par exemple, le code suivant affichera le lien vers le forum uniquement si celui
+
+.. attention::
+
+    le templatetag ``feminize`` est internationalisé.
```
Si j'en crois comment [c'est codé](https://github.com/artragis/zds-site/blob/zep_12_double_stack/zds/utils/templatetags/feminize.py) l'internationalisation ne devrait pas marcher à cause de `return _(article)`. Si tu demande de générer le fichier de traductions, il ne contiendra pas les éléments de ton dictionnaire.

Pour que ce soit internationalisé tu dois faire un truc dans le style de : 

```
articles = {
    "le": _("la"),
    "un": _("une"),
    "Nouveau": _("Nouvelle"),
    "Ce": _("Cette")
}
```

firm1 commented on doc/source/front-end/template-tags.rst:
```diff
@@ -505,3 +480,73 @@ Par exemple, le code suivant affichera le lien vers le forum uniquement si celui
+
+
+    {% load feminize %}
+    {{ "le"|feminize:"partie" }} partie <!-- affiche "la partie" si vous êtes en langue française -->
```
Peut-être mettre une note pour préciser que ce templatetag est sensible à la casse et que si je mets quelque chose comme `{{ "Le"|feminize:"partie" }} partie`, ça ne fonctionne pas ?

pierre-24 replied:
Fait.

firm1 commented:
C'est dommage que github ne permet pas d'afficher tous les fichiers pour le comparatif, je ne peux pas vraiment faire de revue de code :( surtout que ce sont les parties interessantes (les fichiers python) qui ne sont pas affichées

artragis committed:
enlève le trop plein de lignes blanches.

artragis committed:
redirige les vieilles beta.

artragis committed:
Merge pull request #307 from pierre-24/zep_12_link_with_quote

Met le lien vers le message avec la citation

pierre-24 committed:
Redonne à l'auteur le droit de télécharger la source markdown

pierre-24 committed:
Permet de passer à la version en ligne à partir du brouillon (fix #316)

pierre-24 committed:
Fix incohérence de terme sur la page d'aide (fix #312)

pierre-24 committed:
Fix incohérence de vocabulaire sur recherche (fix #311)

pierre-24 committed:
Nettoie commentaires inutiles

pierre-24 committed:
Corrige les `tearDown()`

pierre-24 committed:
Met en évidence `<em>` au lieu de `<i>`

pierre-24 committed:
Ré-insère le `secure` dans `.travis.yml`

pierre-24 committed:
Enlève le markdown des titres des sections des articles (fix #299)

pierre-24 committed:
Si le sujet de bêta est déplacé, un nouveau est créé (fix #310)

pierre-24 committed:
Correction documentation

pierre-24 committed:
Corrige l'internationalisation de `feminise`

pierre-24 committed:
Corrections pour Travis

pierre-24 committed:
Corrige erreur si pas de sujet de bêta

artragis committed:
Merge pull request #318 from pierre-24/zep_12_many_fixes_at_once

Plein de corrections

pierre-24 committed:
Utilise le bon champ lors de l'importation (fix #322)

pierre-24 committed:
Tutoriel -> contenu (fix #324)

pierre-24 committed:
Migration de la validation

pierre-24 committed:
Prot?ge l'importation contre les erreurs d'encodage (fix #321)

pierre-24 committed:
Remet a la page 1 si changement de filtre sur l'aide (fix #320)

pierre-24 committed:
SEO: nouveaux contenus dans le sitemap (fix #319)

+ blacklist des urls de failback des anciens modules dans le robot.txt

pierre-24 committed:
Empeche les admins de lister les contenus en redaction (fix #317)

pierre-24 committed:
Date de publicationd dans le PDF (fix #303)

pierre-24 committed:
Corrige (encore) le .travis.yml

pierre-24 committed:
Renseigne le changement de licence (fix #323)

pierre-24 committed:
Merge pull request #329 from pierre-24/zep_12_migrate_sha_validation

Migration de la validation

pierre-24 committed:
Merge pull request #327 from pierre-24/zep_12_import_use_text_field

Utilise le bon champ lors de l'importation (fix #322)

pierre-24 committed:
Correction license -> licence et test find:*

pierre-24 committed:
Merge pull request #330 from pierre-24/zep_12_unicode_bis

Protège l'importation contre les erreurs d'encodage (fix #321)

artragis committed:
Merge pull request #331 from pierre-24/zep_many_things_at_once_2

Encore une PR fourre-tout

artragis committed:
permet de mettre le +1 en vert lorsqu'on a voté (fix #325)

pierre-24 committed:
Supprime l'auteur meme si il ne fait pas partie de la gallerie

pierre-24 committed:
Assainit le depot avant la copie, puis le clone (fix #284)

pierre-24 committed:
Sauve le commit nettoyage dans l'ancien module aussi

pierre-24 committed:
Supprime `--copy-repository` du update.md

artragis committed:
Merge pull request #332 from pierre-24/zep_12_clone_things_and_clean

Nettoie les erreurs du passé

pierre-24 committed:
Pagination (fix #313, rip performances)

pierre-24 committed:
Trie par date de publication (et non par pk)

artragis committed:
Merge pull request #333 from pierre-24/zep_12_paginate_articles

Pagination des articles

artragis committed:
met la bonne url pour la page d'aide (fix #337)

artragis committed:
enlève 'sommaire du tutoriel' pour un article (fix #340)

artragis committed:
Merge branch 'zep_12_double_stack' of https://github.com/artragis/zds-site into zep_12_double_stack

artragis committed:
redonne les droits au staff.

artragis committed:
évite les pk mal formées (fix #349, #348)

artragis committed:
lance une 404 si l'objet n'existe pas (fix #348)

artragis committed:
permet d'afficher l'éditeur (fix #352)

pierre-24 committed:
Besoin → type sur mobile (fix #354)

pierre-24 committed:
Éditer un commentaire résout également l'alerte (fix #351)

(et fix #347)

pierre-24 committed:
"Dernière mise à jour" (fix #350)

pierre-24 committed:
Empêche les logos de trop grande taille (fix #338)

pierre-24 committed:
Gagne de la place lors de l'affichage d'une partie publique (fix #345)

pierre-24 committed:
Merge pull request #335 from Situphen/zep-12-doc

Quelques corrections dans la documentation et le fichier `update.md`

artragis committed:
corrige une erreur 500, et un todo (fix #359)

et économise une requête un peu bête.

artragis committed:
Merge pull request #355 from pierre-24/zep_12_many_fix_at_once_3

"Many things at Once III": le retour de la vengeance de la correction de bugs en masse

pierre-24 committed:
Permet apercu si pas de JS (fix #343)

pierre-24 committed:
Tutoriel → Contenu dans formulaire de création (fix #356)

pierre-24 committed:
Corrige condition affichage intro/conclu (fix #358)

pierre-24 committed:
Empêche les slugs vides (fix #360)

pierre-24 committed:
Rajoute la "barre jaune" en édition (fix #361)

pierre-24 committed:
Citation sans JS (fix #314)

pierre-24 committed:
Améliore apparition de la "barre jaune" sur commentaires

pierre-24 committed:
Ajoute TU(s) : prévisualisation et citation (avec et sans JS)

artragis committed:
Merge pull request #363 from pierre-24/many_fixes_at_once_4

"Many fixes at once 4": on ne faiblit pas

artragis committed:
fix #362

pierre-24 committed:
Empêche les slugs non-autorisés (fix #360 définitivement)

pierre-24 committed:
Licence par défaut à la création d'un contenu (fix #357)

artragis commented:
Juste une question : quand mergera-t-on cette PR?

Questions subsidiaires @SpaceFox @devock : 
- quand ferons-nous une release complète avec la zep-12?
- quand l'ancienne prod deviendra-t-elle la nouvelle preprod?
- quand aurez-vous le temps de synchroniser les tutos de la prod sur la preprod (que ça soit l'actuelle ou pas)?

SpaceFox commented:
Quand elle sera prête ?  Je n'ai pas de visibilité là-dessus, j'ai entendu parler de corrections de bugs il y a peu.

Réponses subsidiaires :
1. Oui.
2. Quand on aura les accès.
3. Ça n'avais pas été fait il y a peu ?

artragis commented:
Elle est prête amha. Il reste deux "bugs"
- les dates sont en anglais lorsque le pdf est généré automatiquement; 
  On ne sait pas si ça vient vraiement de nous,
- un très léger bug graphique sur mobile (décallage de quelques pixel du 
  bouton "apperçu" quand on répond à une note, apparamment, c'est un bug 
  de l'upstream en plus)

Demandons confirmation @pierre-24 et à nos testeurs @vayel 
@GerardPaligot @sandhose @DevHugo @situphen @firm1.

Pour ce qui est de la synchronisation avec la prod, d'après ce que je 
sais elle est très très très ancienne. Faut demander à @sandhose c'est 
lui qui wipe le plus souvent.

Le 07/09/2015 13:47, SpaceFox a écrit :

> Quand elle sera prête ? Je n'ai pas de visibilité là-dessus, j'ai 
> entendu parler de corrections de bugs il y a peu.
> 
> Réponses subsidiaires :
> 1. Oui.
> 2. Quand on aura les accès.
> 3. Ça n'avais pas été fait il y a peu ?
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/zestedesavoir/zds-site/pull/2956#issuecomment-138277627.

GerardPaligot commented:
Quand est-il du bug de la recherche ?

artragis commented:
On sait toujours pas ce qui coince ni si ce n'est pas à cause des 
problèmes de l'actuel sauvegarde de la béta qui est quelque peu cassée.

Le 07/09/2015 14:10, Gérard Paligot a écrit :

> Quand est-il du bug de la recherche ?
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/zestedesavoir/zds-site/pull/2956#issuecomment-138280894.

firm1 commented:
Personnellement, sur cette zep, voici ce j'ai déjà pu faire : 

## réalisé
- relecture de la doc, du fichiers settings et du fichier tox.ini
- test en mode "utilisateur" sur la base de ce qui est proposé sur la preprod : je teste le front (sur pc et sur mobile) tout comme le back. Quand je tombe sur un bug, c'est reporté sur le dépot d'artragis directement.

## non réalisable
- code-review python : la zep touchant a trop de fichiers, github ne les affiche pas tous dans l'interface de comparaison. ce qui m'empêche de faire une bonne lecture.
- tests en profondeur du module des articles : à cause de [ce bug](https://github.com/artragis/zds-site/issues/360) dont la correction n'a pas été poussée sur la preprod encore, je n'ai pas encore testé l'ensemble du module des articles
- les nouvelles commandes : je n'ai pas testé les nouvelles commandes (recherches, migrations de tutoriels, etc.)

Voili voulou pour ma part.

artragis commented:
> les nouvelles commandes : je n'ai pas testé les nouvelles commandes (recherches, migrations de tutoriels, etc.)

c'est testé dans la procédure de mise en préprod.

> tests en profondeur du module des articles : à cause de ce bug dont la correction n'a pas été poussée sur la preprod encore, je n'ai pas encore testé l'ensemble du module des articles

il me semblait pourtant que ça avait été poussé. faut demander @sandhose il me semble de faire un simple pull.

firm1 commented:
> il me semblait pourtant que ça avait été poussé.

J'ai toujours l'erreur 500 perso

artragis commented:
Ah oui, faut wiper car tu as cassé ton interface :p

Donc il faut refare une migration complète @sandhose

Par ce que j'ai vérifié sur mon instance, le bug est corrigé;

Le 07/09/2015 14:52, firm1 a écrit :

> ```
> il me semblait pourtant que ça avait été poussé.
> ```
> 
> J'ai toujours l'erreur 500 perso
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/zestedesavoir/zds-site/pull/2956#issuecomment-138290486.

artragis commented:
@firm1 tu peux retenter sur la preprod et vérifier que tout va bien.

firm1 commented:
C'est corrigé, mais visiblement [pas comme il faut](https://github.com/artragis/zds-site/issues/365).

artragis commented:
au contraire c'est tout à fait corrigé comme il faut.

artragis committed:
corrige une typo de vocabulaire (fix #366)

GerardPaligot commented:
Ce qu'il faudrait faire pour valider les derniers points, c'est une simulation d'une mise en prod sur la bêta ? (genre avec les données réelles, etc.)

artragis commented:
yep il faut que quelqu'un qui a les droits d'écriture sur la preprod réussie à faire une synchronisation depuis la prod vers la preprod.
Ensuite on lance le script de migration.

artragis committed:
enlève toute évocation à "du tutoriel" (fix #367)

artragis committed:
permet d'éviter l'import de slug mauvais

artragis committed:
affiche le message d'erreur pour une meilleure lisibilité

artragis committed:
typo

artragis committed:
nouvelle typo

artragis committed:
pep8

artragis commented:
Je pense qu'on pourra merger la PR dans la journée. il ne reste plus qu'à valider le dernier bugfix sur l'importation.

Eskimon commented:
\o/

artragis commented:
@sandhose  tu peux refaire un wipe histoire que @firm1  puisse valider tout ça?

GerardPaligot commented:
On doit pas faire une synchro preprod/prod pour valider cette PR ?

artragis commented:
je pensais que ça se ferait après le merge.

GerardPaligot commented:
Moi ça me va !

artragis commented:
ok donc y'a plus qu'à réveiller l'assoc pour migrer la preprod, quelqu'un est en contact avec @devock?
Sinon @SpaceFox @firm1 @ShigeruM (pour la com') @DevHugo @GerardPaligot (parce qu'il ne doit pas avoir vu ce fil :p) @pierre-24 à vous les studios !

firm1 commented on .gitignore:
```diff
@@ -6,13 +6,19 @@ base.db
+/contents-public
 /media
 /articles-data
+contents-private/
```
Pourquoi on a deux fois l'exclusion de contents-private et contents-public. C'est un peu _too much_

firm1 commented on .travis.yml:
```diff
@@ -10,9 +10,11 @@ env:
   matrix:
-    - TEST_APP="-e back_mysql"
+    - TEST_APP="-e back_mysql -- zds.member zds.mp zds.utils zds.forum zds.gallery zds.pages zds.search"
+    - TEST_APP="-e back_mysql_tuto -- zds.article zds.tutorial zds.tutorialv2"
```
Un fois `[testenv:back_mysql_tuto]` suppirmé dans le tox.ini, remplacer `TEST_APP="-e back_mysql_tuto -- zds.article zds.tutorial zds.tutorialv2"` par `TEST_APP="-e back_mysql -- zds.article zds.tutorial zds.tutorialv2"`

firm1 commented on .travis.yml:
```diff
@@ -10,9 +10,11 @@ env:
     - secure: "azmDZZQZzf88zpbkYpLpxI66vpEVyv+kniW0QdWAt4qlys8P5OcO3VJBR5YT85vlvnjN9b6raWQAL1ymee0WmVdTmzXed8XjZv7t9QXVw7pfezxMKlEftVp/4Cu4wtvbew0ViZXNWV2hNXHzEqlhgnoIOq94i0UzZ7grMrI0xm0="
   matrix:
-    - TEST_APP="-e back_mysql"
+    - TEST_APP="-e back_mysql -- zds.member zds.mp zds.utils zds.forum zds.gallery zds.pages zds.search"
```
ici il manque `zds.featured` dans la liste des paramètres

artragis committed:
Merge pull request #373 from artragis/artragis-patch-4

Artragis patch 4

Commit:
vocabulaire

Commit:
Corrige l'importation quand le titre est mauvais

Commit:
oh la vilaine typo

GerardPaligot commented:
On aimerait sans doute l'intervention d'un DTC sur cette PR @SpaceFox.

Accord du merge, tests à venir sur la preprod, etc.

pierre-24 committed:
PEP-8 !

pierre-24 committed:
PEP-8 !

pierre-24 committed:
Corrige doublon dans .gitignore

pierre-24 committed:
Corrige doublon dans .gitignore

pierre-24 committed:
Correction tox.ini et .travis.yml pour les tests unitaires

pierre-24 committed:
Correction tox.ini et .travis.yml pour les tests unitaires

artragis committed:
ajoute un test unitaire sur le cas d'échec

artragis committed:
Merge branch 'zep_12_double_stack' of https://github.com/artragis/zds-site into zep_12_double_stack

artragis committed:
pep8

artragis committed:
Merge branch 'zep_12_double_stack' of https://github.com/artragis/zds-site into zep_12_double_stack

Eskimon commented:
Bon, tout les indicateurs sont au vert, en live d'IRC @artragis me confirme qu'on est bon, du coup je vais déclencher l'apocalypse...

**3**...
**2**...
**1**...
Click...

Situphen commented:
![Bam !](http://jkbaker.com/gallery/d/583-1/nuke.gif)

WinXaito commented:
Applauso !!
